### PR TITLE
chore(javascripts): add eslint no-var

### LINKED ISSUE #1514: resolve no-var violations in javascripts/projectsService.js
We have some very old ESLint rules that I'd like to clean up, and our use of `var` is one of them. ES6 introduces `let` and `const` as a way to make clearer when a variable can be used, and I'd like to reduce our use of `var` where we don't need it:

If you know a bit about JS you can help out by following these steps:

 - create a new branch from `gh-pages`
 - in `javascripts/projectsService.js` add this comment at the top of the file, before any source code:

```js
/* eslint no-var: [ "error" ] */
```

 - in a terminal, run `npm run lint`, which should emit a bunch of violations:

```
$ npm run lint

> up-for-grabs.net@1.0.0 lint /Users/shiftkey/src/up-for-grabs.net
> eslint javascripts/*.js tests/**/*.js


/Users/shiftkey/src/up-for-grabs.net/javascripts/projectsService.js
    8:3  error  Unexpected var, use let or const instead  no-var
   12:3  error  Unexpected var, use let or const instead  no-var
   ...
```

 - run `npm run lint-fix` to fix whatever can be automatically tidied up
 - review and address the remaining violations manually
 - run `npm run prettier-fix` to ensure the formatting of the changes is fine
 - [test the site locally](https://github.com/up-for-grabs/up-for-grabs.net#testing-the-site-locally)
 - commit the changes and open a PR
 - test out the Netlify deploy site (click the `Details` link) and confirm the client-side functionality is still working

### PR DESCRIPTION
**Pull request by firfircelik**
Resolves https://github.com/up-for-grabs/up-for-grabs.net/issues/1514

### CODE REVIEW

**Comment by firfircelik** [comment]
@shiftkey I don't get it, lint issue still cont. when I run after lint-fix and prettier.

**Comment by shiftkey** [comment]
@firfircelik are these the two errors you are seeing after running `npm run lint-fix`?

```

/Users/shiftkey/src/up-for-grabs.net/javascripts/projectsService.js
   14:3  error  Unexpected var, use let or const instead             no-var
  137:9  error  '_tagsMap' is never reassigned. Use 'const' instead  prefer-const

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by firfircelik** [comment]
ok right now I see this error


```
> up-for-grabs.net@1.0.0 lint-fix /home/firat/up-for-grabs.net/up-for-grabs.net
> eslint --fix javascripts/*.js tests/**/*.js


Oops! Something went wrong! :(

ESLint: 6.5.1.

ESLint couldn't find the config "prettier" to extend from. Please check that the name of the config is correct.

The config "prettier" was referenced from the config file in "/home/firat/up-for-grabs.net/up-for-grabs.net/.eslintrc.json".

If you still have problems, please stop by https://gitter.im/eslint/eslint to chat with the team.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! up-for-grabs.net@1.0.0 lint-fix: `eslint --fix javascripts/*.js tests/**/*.js`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the up-for-grabs.net@1.0.0 lint-fix script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by shiftkey** [comment]
@firfircelik you need to run `npm install` to install the missing dependencies that `eslint` now needs

**Comment by firfircelik** [comment]
@shiftkey ok I found the problem, handlebars version need to be 4.4.3. If its good to you I will update it too

**Comment by shiftkey** [comment]
> ok I found the problem, handlebars version need to be 4.4.3. If its good to you I will update it too

Don't worry about that problem - it's a dependency of our test runner and a low-risk issue, and can get addressed when the upstream dependencies upgrade.

File: javascripts/projectsService.js [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
 /* eslint implicit-arrow-linebreak: [ "off" ] */
 /* eslint no-confusing-arrow: [ "off" ] */
+/* eslint no-var: [ "error" ] */
 
 // @ts-nocheck
 
 // required for loading into a NodeJS context
 if (typeof define !== 'function') {
-  var define = require('amdefine')(module);
+  define = require('amdefine')(module);
```
Comment (shiftkey) [inline javascripts/projectsService.js]: This is an exception that we need to figure out how to handle better:

```suggestion
  /* eslint-disable-next-line no-var */
  var define = require('amdefine')(module);
```

**Comment by firfircelik** [comment]
@shiftkey I think its GTM

File: javascripts/projectsService.js [hunk 2]
```diff
@@ -132,27 +133,27 @@ define(['underscore'], _ => {
...
-      var _entry = _tagsMap[tagLowerCase];
+      const _entry = _tagsMap[tagLowerCase];
       _entry.frequency++;
       _entry.projects.push(projectName);
     };
 
     this.getTagsMap = function() {
       // https://stackoverflow.com/questions/16426774/underscore-sortby-based-on-multiple-attributes
-      return (_orderedTagsMap =
+      return (const_orderedTagsMap =
```
Comment (shiftkey) [inline javascripts/projectsService.js]: This typo needs to be undone

```suggestion
      return (_orderedTagsMap =
```

File: javascripts/projectsService.js [hunk 3]
```diff
@@ -132,27 +133,27 @@ define(['underscore'], _ => {
     return _.flatten(results, (arr1, arr2) => arr1.append(arr2));
   };
 
-  var TagBuilder = function() {
-    var _tagsMap = {},
+  const TagBuilder = function() {
+    const _tagsMap = {},
```
Comment (shiftkey) [inline javascripts/projectsService.js]: Let's split these two up so they can be assigned separately:

```suggestion
    const _tagsMap = {};
```

File: javascripts/projectsService.js [hunk 4]
```diff
@@ -132,27 +133,27 @@ define(['underscore'], _ => {
...
       _orderedTagsMap = null;
```
Comment (shiftkey) [inline javascripts/projectsService.js]:
```suggestion
      let _orderedTagsMap = null;
```

**Review by shiftkey — CHANGES_REQUESTED** [review]
@firfircelik thanks for working through this.

I think we're getting close - just a bit of tidy-up left over on these corner cases.

File: javascripts/projectsService.js (hunk 4 above)
Comment (firfircelik) [reply javascripts/projectsService.js]: all done

**Comment by shiftkey** [comment]
@firfircelik changes look good - I just rebased and amended the commit because:

 - it referenced an old PR, potentially confusing others when it gets merged
 - it was authored by the `@github-actions` account, and not you

Should be good to merge when this goes green again.

**Review by shiftkey — APPROVED** [review]

**Pull request merged**